pkg/scheduler/core/generic_scheduler.go

```diff
@@ -167,7 +167,6 @@ type genericScheduler struct {
 	pvcLister                corelisters.PersistentVolumeClaimLister
 	pdbLister                algorithm.PDBLister
 	disablePreemption        bool
-	lastIndex                int
```

Comment:
I moved the declaration of lastIndex into genericScheduler#findNodesThatFit
I also let AllNodes make copy of nt.allNodes
```
func (nt *NodeTree) AllNodes() []string {
	nt.mu.RLock()
	defer nt.mu.RUnlock()
	var nodes []string
	nodes = append(nodes, nt.allNodes...)
	return nodes
}
```
I wonder if the above is enough to handle the race condition.